stock_mts_mto_mrp_rule/models/stock_move.py

```diff
@@ -0,0 +1,45 @@
+from odoo import models
+from odoo.tools import float_compare, float_is_zero
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _adjust_procure_method(self):
+        precision = self.env["decimal.precision"].precision_get(
+            "Product Unit of Measure"
+        )
+        for move in self:
+            product_id = move.product_id
+            domain = [
+                ("location_src_id", "=", move.location_id.id),
+                ("location_id", "=", move.location_dest_id.id),
+                ("action", "!=", "push"),
+            ]
+            rules = self.env["procurement.group"]._search_rule(
+                False, product_id, move.warehouse_id, domain
+            )
+            if not rules or rules and rules.action != "split_procurement":
+                return super(StockMove, self)._adjust_procure_method()
+            needed_qty = rules.get_mto_qty_to_order(
+                product_id, move.product_qty, move.product_uom, {}
+            )
+            if float_is_zero(needed_qty, precision_digits=precision):
+                move.procure_method = "make_to_stock"
+            elif (
+                float_compare(needed_qty, move.product_qty, precision_digits=precision)
+                == 0.0
+            ):
+                move.procure_method = "make_to_order"
+            else:
+                mts_qty = move.product_uom_qty - needed_qty
+                move.copy(
```

Comment:
You do nothing with new copied move ? Is this really a copy or do you want to split quantities (as there is a split method) ?

Reply:
Resolved. I'm applied '_split' method from stock.move.